rmw_fastrtps_cpp/src/TypeSupport.cpp

```diff
@@ -202,8 +202,10 @@ bool TypeSupport::serializeROSmessage(eprosima::fastcdr::Cdr &ser, const rosidl_
             {
                 case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BOOL:
                     {
-                        bool sb = false, &b = *(bool*)field;
-                        if(b) sb = true; 
+                        bool sb = false;
+                        // don't cast to bool here because if the bool is
+                        // uninitialized the random value can't be deserialized
+                        if(*(uint8_t*)field) sb = true;
```

Comment:
I could imagine that `*(bool *)field` could result in a bad access if `field` points to memory outside of the process. I could also imagine that copying the memory contents at `field` into a bool on the stack could mess up the stack, but I wouldn't think that's possible since the space reserved for `b` on the stack would be the same size as the memory copied from `field` and therefore the bogus contents of memory at `field` should not "leak" into the stack. I guess the compiler could be generating slightly different code here as an optimization which would make that possible.

Maybe I'm completely off-base, I'm just thinking out loud.